### PR TITLE
Refactor alter table

### PR DESCRIPTION
**Pull request by dhmlau**
### Description
Refactor alter table

#### Related issues
Related to https://github.com/strongloop/loopback-connector/issues/15

### Checklist

- [ ] New tests added or existing tests modified to cover all changes
- [ ] Code conforms with the [style
  guide](http://loopback.io/doc/en/contrib/style-guide.html)

### CODE REVIEW

**Comment by ssh24** [comment]
@slnode test please

File: lib/migration.js [hunk 1]
```diff
@@ -138,15 +127,105 @@ function mixinMigration(MsSQL) {
...
+      });
+
+      if (columnsToDrop.length) {
+        statements.push('DROP COLUMN' + columnsToDrop.join(',' + MsSQL.newline));
+      }
+    };
+    return statements;
+  };
+
+  MsSQL.prototype.addIndexes = function(model, actualIndexes, cb) {
```
Comment (jannyHou) [inline lib/migration.js]: Does it need a cb? it's not called inside function and the caller doesn't provide it either.

File: lib/migration.js [hunk 2]
```diff
@@ -88,47 +88,36 @@ function mixinMigration(MsSQL) {
...
-          };
-        }
-        ai[name].columns[i.Seq_in_index - 1] = i.Column_name;
-      });
-    }
-    var aiNames = Object.keys(ai);
+  MsSQL.prototype.getColumnsToAdd = function(model, actualFields) {
+    var self = this;
+    var m = self._models[model];
+    var propNames = Object.keys(m.properties);
```
Comment (jannyHou) [inline lib/migration.js]:
```js
var propNames = Object.keys(m.properties).filter(function(name) {
      return !!m.properties[name];
    });
```

File: lib/migration.js [hunk 3]
```diff
@@ -88,47 +88,36 @@ function mixinMigration(MsSQL) {
...
     // change/add new fields
     propNames.forEach(function(propName) {
       if (propName === idName) return;
       var found;
-      actualFields.forEach(function(f) {
-        if (f.Field === propName) {
-          found = f;
-        }
-      });
+      if (actualFields) {
```
Comment (jannyHou) [inline lib/migration.js]: hmm, I might miss something, why do we add a check `if (actualFields)` here?

Comment (jannyHou) [inline lib/migration.js]: A second thought, it's good to make the check :)

**Comment by ssh24** [comment]
@slnode test please

**Pull request merged**